### DISCUSSION TITLE
Naming convention for admin views

### DISCUSSION THREAD

**Discussion by AndiLeni** [joomengine/Joomla-Component-Builder#796] [Q&A]
Hello,

I just started using JCB, and it looks pretty impressive.
A problem I faced while creating admin views was the naming convention.
I have a `Name (single record)` set to "Angebot" ('Offer' in german).
When I set the `Name (list of records)` to "Angebote" (correct plural of 'Offers' in german) the list view in my component is not working.
When I set it to "Angebots" the list view is working fine.

Is this a bug or a feature I am not aware of?

Thanks and kind regards
Andreas

**Comment by amolsawant1989** [comment]
Hi AndiLeni,

I think It's not a bug at all it's just a standard naming conventions which all of us follow.. because it's written in English so for plural it usages "s". so don't worry use it with "S".....

**Reply by AndiLeni** [reply]
Thanks for the answer. 
I am fine using it in this way. 
It's just somehow unpleasant to read it in German, since this plural is grammatically wrong.
I am not aware of other languages and their grammar, but I am sure this can happen im many languages.

**Reply by outsch** [reply]
Hi AndiLeni,
it may be a misperception on the function of the admin view, since it only labels such elements in your development. When using these views (admin or site views alike) in the later application, you may give them properly written labels in German or any other language, so there is not need to show somthing ugly like "Angebots" in your later application.
So even if on my below example my "development-names" are "manufacturer" (singluar) and "manufacturers" (plural), in the application they appear properly as "Hersteller":
![image](https://user-images.githubusercontent.com/46380117/130968027-d384a1f4-e6a0-44f1-8181-535ef2e25654.png)

So I would recommend that you stick to plain English for your entire development terminology. This is fully decoupled from how these items appear later in the application.

**Reply by AndiLeni** [reply]
@outsch Thanks for the answer.
Unfortunately I could still not figure out how to translate those labels:
![grafik](https://user-images.githubusercontent.com/13385771/132994064-d112bf5e-5ead-4f9c-80ea-373ee18ae5c5.png)

The label in the "Components" dropdown menu is correct, but the labels on the components admin page are the ones I used in english.

![grafik](https://user-images.githubusercontent.com/13385771/132994166-c782686f-84e0-4b73-acae-257758042b5f.png)

**Comment by Llewellynvdm** [comment]
### Total Freedom
The **s** should not be required, as JCB can build in other languages, I know many who use German, France and Dutch as the main language. You will see that you can update the main language in the global options tab->Compiler:
![image](https://user-images.githubusercontent.com/5607939/131117850-837e7328-fef1-4af9-8375-e424d0cc4926.png)

These tutorials should also help:
- https://www.youtube.com/watch?v=zzAcVkn_cWU

### This means

You can use any names as long as they are not the same, and not used twice in the same component. There are a few other constraints, read here: https://docs.joomla.org/J3.x:File_Structure_and_Naming_Conventions

> let me know if you have any further questions.

**Reply by Llewellynvdm** [reply]
Do note that if you use another language, then English, you must set it globally, and it will affect all your components on this website..

So each install of JCB can just run/build for one main language. Setting it globally will make sure the [correct conversions](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/helpers/componentbuilder.php#L717) are used. Also make the  Field Name Builder to be set to Alphanumeric:
![image](https://user-images.githubusercontent.com/5607939/131119260-0d54be16-e8f5-4f07-b823-63a9453f2b5a.png)

**Reply by AndiLeni** [reply]
Thanks for the quick answer.
I will give it a try and see if I can change it to fit my needs :-)
